qa/qa-captain/bashrc

```diff
@@ -0,0 +1,1116 @@
+# ======================================================================================
+#
+# I highly encourage you to play with renaming functions and aliases to your liking.
+# What you type and call should coincide with your internal monologues expectations. 
+# Some of the functions like get.k3s and get.rke2 are redundant just to meet the expectations
+# of my internal monologue when typing. You could just create a new function called get.product or 
+# get.cli which expects you to pass in the name of the product k3s or rke2.
+#
+# ======================================================================================
+# ================================ ~ CLUSTERS ~ ========================================
+# ======================================================================================
+
+
+# --- download k3s install.sh ---
+function get.k3s() {
+    curl https://get.k3s.io --output install-k3s.sh
+    sudo chmod +x install-k3s.sh
+}
+
+# --- install k3s server or agent and run---
+function go.k3s() {
+    local VERSION="${1}"
+    #local CHANNEL="${3:-testing}"
+    #local TYPE="${4:-server}"
+    sudo INSTALL_K3S_VERSION="$VERSION" INSTALL_K3S_EXEC=server ./install-K3s.sh
+    #sudo INSTALL_"$PRODUCT"_VERSION="$VERSION" INSTALL_"$PRODUCT"_CHANNEL="$CHANNEL" INSTALL_"$PRODUCT"_EXEC="$TYPE" ./install-"$PRODUCT".sh
+}
+
+# --- download rke2 install.sh ---
+function get.rke2() {
+    curl https://get.rke2.io --output install-rke2.sh
+    sudo chmod +x install-rke2.sh
+}
+
+# --- start rke2 systemctl service type ---
+function go.rke2() {
+    local TYPE="${1:-server}"
+    sudo systemctl enable rke2-"$TYPE" --now
+}
+
+# --- print to console current config file ---
+function get.figs() {
+    local PRODUCT="${1:-k3s}"
+    printf '=========== %s config =========== \n' "$PRODUCT"
+    sudo cat /etc/rancher/"$PRODUCT"/config.yaml;
+}
+
+# --- restore previously saved config file (from void function) ---
+function go.replay() {
+    # after you've called void the tmp-confs directory is made with your previos config
+    local PRODUCT="${1:-k3s}"
+    sudo mkdir -p /etc/rancher/"$PRODUCT"/;
+    sudo cp ~/tmp-confs/"$PRODUCT"-config.yaml /etc/rancher/"$PRODUCT"/config.yaml
+}
+
+# --- set KUBECONFIG environment variable ---
+function set.kubefig() {
+    local PRODUCT="${1:-k3s}"
+    export KUBECONFIG=/etc/rancher/"$PRODUCT"/"$PRODUCT".yaml
+    sudo chmod 644 /etc/rancher/"$PRODUCT"/"$PRODUCT".yaml
+}
+
+# --- quickly edit product config file ---
+function conf() {
+    local PRODUCT="${1:-k3s}"
+    sudo vi /etc/rancher/"$PRODUCT"/config.yaml
+}
+
+# --- prints out node token ---
+function get.token() {
+    local PRODUCT="${1:-k3s}"
+    sudo cat /var/lib/rancher/"$PRODUCT"/server/node-token
+}
+
+# --- killall uninstall all ---
+function void() {
+    # sudo systemctl list-units | grep -i -e k3s -e rke2 ??? find whats installed - then just remove those instead of trying all
+    mkdir ~/tmp-confs/
```

Comment:
Can we use `mktemp` here?

Reply:
I've been leaving the config file in the user directory for QA to have an additional opportunity to edit the file before quickly launching a new installation re-using the previous config.yaml